### PR TITLE
adr: Pull Request and Code Review Guide

### PR DESCRIPTION
**Pull request by irq0**

### CODE REVIEW

**Comment by giubacc** [comment]
Good one.
Should we add something specific regarding how to handle a dispute ?
Who should be the ultimate decider when there is a divergence of views?

**Comment by jecluis** [comment]
> Good one. Should we add something specific regarding how to handle a dispute ? Who should be the ultimate decider when there is a divergence of views?

Hopefully that won't happen often, or at all, as we are all adults and should be able to concede our position during a discussion. But if it comes to a point where such a thing is needed, I'm sure either Jesus, Lars, or I, could own the decision.

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (jecluis) [inline docs/decisions/0008-pull-requests-code-review-guide.md:16]: Okay, I'd add a caveat here: if any one of the reviewers feels strongly about something, it would be good practice to check with them before merging the PR. Having one reviewer giving the thumbs up and another feeling strongly about a thumbs down, and merging nonetheless, is a bit of a dick move ;)

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (jecluis) [inline docs/decisions/0008-pull-requests-code-review-guide.md:48]: Maybe add that individual commits need to be GPG signed and have a DCO?

**Comment by giubacc** [comment]
> > Good one. Should we add something specific regarding how to handle a dispute ? Who should be the ultimate decider when there is a divergence of views?
> 
> Hopefully that won't happen often, or at all, as we are all adults and should be able to concede our position during a discussion. But if it comes to a point where such a thing is needed, I'm sure either Jesus, Lars, or I, could own the decision.

Fair, I think this is an important rule that should be coded into the ADR.

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (jecluis) [inline docs/decisions/0008-pull-requests-code-review-guide.md]: I'd add "productive, constructive feedback". Just being "productive" is not enough for a civil discussion :)

**Review by jecluis — DISMISSED** [review]
Not holding the PR on the account of my comments, but would love to have them addressed :)

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (jecluis) [reply docs/decisions/0008-pull-requests-code-review-guide.md:16]: Also, as an example, take this PR: it's something that should be checked out by everyone. I've already signed-off on it. Should we merge it immediately, or wait for more reviews? :)

**Comment by irq0** [comment]
> Should we add something specific regarding how to handle a dispute ?
> Who should be the ultimate decider when there is a divergence of views?

I'd treat dispute as a corner case, that we don't need to optimize. I suggest that on dispute we talk it out in one of the standups.

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (irq0) [reply docs/decisions/0008-pull-requests-code-review-guide.md]: :+1: editing it in

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (irq0) [reply docs/decisions/0008-pull-requests-code-review-guide.md:48]: makes sense. adding it in

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (irq0) [reply docs/decisions/0008-pull-requests-code-review-guide.md:16]: > Also, as an example, take this PR: it's something that should be checked out by everyone. I've already signed-off on it. Should we merge it immediately, or wait for more reviews? :)

Good point. I should have pointed out that this is one of those consensus PRs

**Comment by irq0** [comment]
This would be one of those PRs that needs universal approval

File: docs/decisions/0008-pull-requests-code-review-guide.md
Comment (irq0) [reply docs/decisions/0008-pull-requests-code-review-guide.md:16]: > Okay, I'd add a caveat here: if any one of the reviewers feels strongly about something, it would be good practice to check with them before merging the PR. Having one reviewer giving the thumbs up and another feeling strongly about a thumbs down, and merging nonetheless, is a bit of a dick move ;)

You are right. Good points. I started the doc with "This is a guide not law. Use common sense. Have fun." for that reason. A  plea against dick moves :). 

How about a small addition for disapproval:

 5. Discuss, address feedback until ≥ 1 reviewer(s) approve. No reviewer(s) disapprove.

**Review by votdev — DISMISSED** [review]
LGTM

**Review by giubacc — DISMISSED** [review]

**Comment by jecluis** [comment]
I think that's a good addition, but I'd state "no active reviewer
disapproves". Because I can see a scenario where one goes away on vacation
with a request for changes, and we don't want to hold the pr open forever.
I know, edge case, but I'd rather have some clarity from the start. If
that's too much, I'm okay with the proposed phrasing.

**Comment by irq0** [comment]
Last push has the changes from the discussion with @jecluis above. 
Additions:

Individual commits are GPG signed and have a DCO.
Provide productive, constructive feedback.
No active reviewer disapproves.

**Review by jhmarina — COMMENTED** [review]
lgtm!

**Review by jecluis — APPROVED** [review]

**Review by m-ildefons — APPROVED** [review]

**Review by 0xavi0 — APPROVED** [review]
lg2m

**Pull request merged**
